onnxruntime/core/providers/tensorrt/tensorrt_execution_provider.cc

```diff
@@ -93,17 +93,27 @@ TensorrtExecutionProvider::TensorrtExecutionProvider(const TensorrtExecutionProv
       {OrtMemTypeCPUOutput, [](int) { return onnxruntime::make_unique<CUDAPinnedAllocator>(0, TRT_PINNED); }, std::numeric_limits<size_t>::max()});
   InsertAllocator(CreateAllocator(pinned_allocator_info, device_id_));
 
-  const char* batch_env = getenv("ORT_TENSORRT_MAX_PARTITION_ITERATIONS");
-  if (batch_env)
-    max_partition_iterations_ = atoi(batch_env);
+  // Get environment variables
+  const Env& env_instance = Env::Default();
+  const std::string max_partition_iterations_env = env_instance.GetEnvironmentVar(tensorrt_env_vars::kMaxPartitionIterations);
+  if (!max_partition_iterations_env.empty()) {
+    max_partition_iterations_ = stoi(max_partition_iterations_env);
+  }
+
+  const std::string min_subgraph_size_env = env_instance.GetEnvironmentVar(tensorrt_env_vars::kMinSubgraphSize);
+  if (!min_subgraph_size_env.empty()) {
+    min_subgraph_size_ = stoi(min_subgraph_size_env);
+  }
 
-  const char* subgraph_size_env = getenv("ORT_TENSORRT_MIN_SUBGRAPH_SIZE");
-  if (subgraph_size_env)
-    min_subgraph_size_ = atoi(subgraph_size_env);
+  const std::string max_workspace_size_env = env_instance.GetEnvironmentVar(tensorrt_env_vars::kMaxWorkspaceSize);
+  if (!max_workspace_size_env.empty()) {
+    max_workspace_size_ = stoi(max_workspace_size_env);
```

Comment:
the configuring environment variables instructions in the TensorRT-ExecutionProvider.md documentation doesn't work then since 2GB is greater than what stoi can accept. 
### override default max workspace size to 2GB
export ORT_TENSORRT_MAX_WORKSPACE_SIZE=2147483648

don't use stoi for workspace size.  or change the units to megabytes rather than bytes.